test.sh

```diff
@@ -1,32 +1,40 @@
-#!/bin/sh -ex
+#!/bin/sh -e
+
+run() {
+    normal='\e[0m'
+    yellow='\e[33m'
+    printf "${yellow}%s${normal}\n" "$*"
+
+    "$@"
```

Comment:
Both `"$*"` and `$*` do wrong things:
- with `"$*"`, the whole command is interpreted as a single word;
- with `$*`, the command gets expanded twice.